view/frontend/templates/button_product_page.phtml

```diff
@@ -0,0 +1,230 @@
+<?php
+/**
+ * Bolt magento2 plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2018 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+/**
+ * Product page Bolt Checkout button
+ *
+ * @var $block \Bolt\Boltpay\Block\JsProductPage
+ */
+if ($block->shouldDisableBoltCheckout()) return;
+
+if (!$block->IsSupportableType()) return;
+
+$trackCallbacks = $block->getTrackCallbacks();
+
+?>
+    <div class="bolt-product-checkout-button bolt-multi-step-checkout <?=$block->getAdditionalCheckoutButtonClass();?>"></div>
+    <script>
+        // Store the configuration parameters passed from the php block
+        // in the global object. Used in this file and on the payment page
+        // in payment method renderer, vendor/boltpay/bolt-magento2/view/frontend/web/js/view/payment/method-renderer/boltpay.js
+        window.boltConfig = <?php echo $block->getSettings(); ?>;
+        require([
+            'jquery',
+            'Magento_Customer/js/model/authentication-popup',
+            'Magento_Customer/js/customer-data',
+            'mage/validation/validation',
+            'mage/cookies'
+        ], function ($, authenticationPopup, customerData) {
+            var customer = customerData.get('customer');
+            var shoppingCart = customerData.get('cart');
+            var $qty = jQuery('#qty');
+            var max_qty = <?= $block->getProductQty(); ?>;
+
+            var trackCallbacks = {
```

Comment:
seems lots of shared code with other file. Would it make sense to combine?

Reply:
I don't mind, but we need to decide together. I don't know much about magento2, but it seems to me that we could move JS code from *.phtml files into php class.
We can create method into php class that will generate JS code according to parameters.
We did it into woocommerce, but it's a big change. It makes it easier to change JS code, but of course, it affects readability.
@daisy1754 @matt-thomason @danac-gs What's your opinion?

Reply:
After the conversation with @matt-thomason I would prefer to put it off and do in future in separate PR